source/appModules/outlook.py

```diff
@@ -427,49 +415,54 @@ def _get_name(self):
 				unread=selection.unread
 			except COMError:
 				unread=False
-			# Translators: when an email is unread
-			if unread: textList.append(_("unread"))
-			try:
-				flagIcon=selection.flagIcon
-			except COMError:
-				flagIcon=0
-			flagIconLabel=oleFlagIconLabels.get(flagIcon)
-			if flagIconLabel: textList.append(flagIconLabel)
-			try:
-				attachmentCount=selection.attachments.count
-			except COMError:
-				attachmentCount=0
-			# Translators: when an email has attachments
-			if attachmentCount>0: textList.append(_("has attachment"))
-			try:
-				importance=selection.importance
-			except COMError:
-				importance=1
-			importanceLabel=importanceLabels.get(importance)
-			if importanceLabel: textList.append(importanceLabel)
 			try:
 				messageClass=selection.messageClass
 			except COMError:
 				messageClass=None
-			if messageClass=="IPM.Schedule.Meeting.Request":
-				# Translators: the email is a meeting request
-				textList.append(_("meeting request"))
+			if self.appModule.outlookVersion<15:
+				# Translators: when an email is unread
+				if unread: textList.append(_("unread"))
+				if messageClass=="IPM.Schedule.Meeting.Request":
+					# Translators: the email is a meeting request
+					textList.append(_("meeting request"))
+			elif messageClass is not None:
+				# Replied or forwarded state for this message is available from the object's value.
+				# We must parse this value correctly, as it may contain redundant information, such as the message class and read value.
+				# We only expose the unread state, and message class for non-messages.
+				# The several states are localized and separated by a space.
+				# Example output: 'Meeting request Replied Read'
+				valueParts = self._get_value().split(" ")
+				valueCount = len(valueParts)
+				# The last valuePart indicates whether the message is read or unread.
+				# Do not expose the read state
+				lastPart = valueCount if unread else valueCount-1
+				# The first valuePart is the type of the selection, e.g. Message, Contact, meeting request.
+				# We can safely assume that the message class of a regular message or appointment is one word.

```

Comment:
Yet the example you give above is "Meeting request". Is this not regular? If so, 2 words should be filtered out.